src/types/types.ts

```diff
@@ -0,0 +1,44 @@
+export interface Post {
+  title: string,
+  body: string,
+  user: User | null,
+  comment: Comment[] | null,
```

Comment:
`comments` property will never be `null` because the `filter` method returns array in any case.